x-pack/plugins/security_solution/server/lib/detection_engine/rule_schema/model/rule_schemas.ts

```diff
@@ -285,20 +285,32 @@ export const internalRuleCreate = t.type({
   enabled: IsRuleEnabled,
   actions: RuleActionArrayCamel,
   params: ruleParams,
+});
+const internalRuleCreateOptional = t.type({
   throttle: t.union([RuleActionThrottle, t.null]),
   notifyWhen,
 });
+export const internalRuleCreate = t.intersection([
+  internalRuleCreateOptional,
+  internalRuleCreateRequired,
+]);
 export type InternalRuleCreate = t.TypeOf<typeof internalRuleCreate>;
 
-export const internalRuleUpdate = t.type({
+const internalRuleUpdateRequired = t.type({
   name: RuleName,
   tags: RuleTagArray,
   schedule: t.type({
     interval: t.string,
   }),
   actions: RuleActionArrayCamel,
   params: ruleParams,
+});
+const internalRuleUpdateOptional = t.type({
   throttle: t.union([RuleActionThrottle, t.null]),
   notifyWhen,
 });
+export const internalRuleUpdate = t.intersection([
+  internalRuleUpdateOptional,
+  internalRuleUpdateRequired,
+]);
```

Comment:
Sorry, I don't see any functional changes in these schemas, they seem to be identical. I think ideally we'd either need to make optional parts in fact optional

```ts
const internalRuleCreateOptional = t.partial({
  throttle: t.union([RuleActionThrottle, t.null]),
  notifyWhen,
});

const internalRuleUpdateOptional = t.partial({
  throttle: t.union([RuleActionThrottle, t.null]),
  notifyWhen,
});
```

or keep the schemas unchanged.

Or is this some kind of preparational refactoring?

Reply:
These are supposed to be `partial`s, thanks for catching that.

Reply:
@Zacqary 👍 

I have a slightly different idea. So these are types that we use on the Security Solution's side for our own safety. Right now on our side, we use only rule-level `notifyWhen` and `throttle` attributes, and we don't use action-level attributes. This means that for us and our users these two should be required until we add support for the action-level attributes to the app. So I'd keep them required and revert changes in this file.

Is there any reason why these two should be optional here? E.g. can the user, using Alerting's HTTP API, set these to undefined?

Reply:
Yes, the Alerting HTTP API does allow them to be `undefined` as of this PR